1.8/overview/concepts.md

```diff
@@ -1,121 +1,282 @@
+
+A Mesos framework consists of a scheduler, its tasks, and optionally custom executors.
+
+- The term framework and scheduler are sometimes used interchangeably. Prefer scheduler within the context of DC/OS.
+- For more information about framework schedulers and executors, see the [Application Framework development guide](http://mesos.apache.org/documentation/latest/app-framework-development-guide/).
+
+### <a name="mesos-role"></a>Role
+
+A Mesos role is a group of Mesos Frameworks that share reserved resources, persistent volumes, and quota. These frameworks are also grouped together in Mesos' hierarchical Dominant Resource Fairness (DRF) share calculations.
+
+- Roles are often confused as groups of resources, because of they way they can be statically configured on the agents. The assignment is actually the inverse: resources are assigned to roles.
+- Role resource allocation can be configured statically on the Mesos agent or changed at runtime using the Mesos API.
+
+### <a name="resource-offer"></a>Resource Offer
+
+A Mesos resource offer provides a set of unallocated resources (e.g. cpu, disk, memory) from an agent to a scheduler so that the scheduler may allocate those resources to one or more tasks. Resource offers are constructed by the leading Mesos master, but the resources themselves are reported by the individual agents.
+
+### <a name="mesos-containerizer"></a>Containerizer
+
+A Mesos containerizer is a containerization and resource isolation abstraction around a specific container runtime, namely the Mesos Universal Container Runtime and the Docker Runtime
+
+#### <a name="mesos-universal-container-runtime"></a>Mesos Universal Container Runtime
+
+The Mesos Universal Container Runtime is a containerizer that supports traditional Mesos containers around binary executables and also Mesos containers launched from Docker images. Containers managed by the Mesos Universal Container Runtime do not use [Docker-Engine](https://www.docker.com/products/docker-engine), even if launched from a Docker image.
+
+#### <a name="mesos-docker-runtime"></a>Docker Runtime
+
+The Docker Runtime is a containerizer that supports launching Docker containers from Docker images with [Docker-Engine](https://www.docker.com/products/docker-engine).
+
+### <a name="mesos-exhibitor-zookeeper"></a>Exhibitor &amp; Zookeeper
+
+Mesos depends on ZooKeeper, a high-performance coordination service to manage the cluster state. Exhibitor automatically configures and manages ZooKeeper on the [master nodes](#master-node).
+
+### <a name="mesos-exhibitor-zookeeper"></a>Mesos-DNS
+
+Mesos-DNS is a DC/OS component that provides service discovery within the cluster. Mesos-DNS allows applications and services that are running on Mesos to find each other by using the domain name system (DNS), similar to how services discover each other throughout the Internet.
+
+- For more information, see the [Mesos-DNS repository](https://github.com/mesosphere/mesos-dns) on GitHub.
```

Comment:
We should probably change this link to: http://localhost:3000/docs/1.8/usage/service-discovery/mesos-dns/